src/pages/workspaces/workspace/tools/WorkflowView.js

```diff
@@ -492,13 +498,27 @@ const WorkflowView = _.flow(
     return this.isSingle() || !!rootEntityType
   }
 
+  loadNewMethodConfig = _.flow(
+    withErrorReporting('Error updating config'),
+    Utils.withBusyState(v => this.setState({ updatingConfig: v }))
+  )(async newSnapshotId => {
+    const { ajax: { Methods } } = this.props
+    const { modifiedConfig: { methodRepoMethod: { methodNamespace, methodName } }, modifiedConfig } = this.state
+    const config = await Methods.template({ methodNamespace, methodName, methodVersion: newSnapshotId })
+    const inputsOutputs = await Methods.configInputsOutputs(config)
+
+    this.setState({ inputsOutputs: _.update('inputs', _.sortBy('optional'), inputsOutputs) })
+    this.setState(_.set(['modifiedConfig', 'methodRepoMethod'], _.merge(modifiedConfig.methodRepoMethod, config.methodRepoMethod)))
```

Comment:
nit: Do we need to `merge` here, or can we just overwrite with the whole new value?